Sources/AblyAssetTrackingPublisher/DefaultPublisher.swift

```diff
@@ -311,36 +285,40 @@ extension DefaultPublisher {
         self.route = event.route
     }
 
-    // MARK: Add trackable
-    private func performAddTrackableEvent(_ event: Event.AddTrackableEvent) {
+    private func performAddOrTrack(_ trackable: Trackable, resultHandler: @escaping ResultHandler<Void>, completion: @escaping () -> Void){
         guard !state.isStoppingOrStopped else {
-            publisherStoppedCallback(handler: event.resultHandler)
+            publisherStoppedCallback(handler: resultHandler)
             return
         }
 
-        guard !trackables.contains(event.trackable) else {
-            let error = ErrorInformation(type: .trackableAlreadyExist(trackableId: event.trackable.id))
-            callback(error: error, handler: event.resultHandler)
+        guard !trackables.contains(trackable) else {
+            completion()
             return
         }
-     
-     
+
+
         ablyPublisher.connect(
-            trackableId: event.trackable.id,
-            presenceData: presenceData,
-            useRewind: false
+                trackableId: trackable.id,
+                presenceData: presenceData,
+                useRewind: false
         ) { [weak self] result in
-     
+
             switch result {
             case .success:
-                self?.enqueue(event: .presenceJoinedSuccessfully(.init(trackable: event.trackable) { [weak self] _ in
-                    self?.callback(value: Void(), handler: event.resultHandler)
+                self?.enqueue(event: .presenceJoinedSuccessfully(.init(trackable: trackable) { [weak self] _ in
+                    self?.callback(value: Void(), handler: resultHandler)
                 }))
             case .failure(let error):
-                self?.callback(error: error, handler: event.resultHandler)
+                self?.callback(error: error, handler: resultHandler)
             }
         }
     }
+    // MARK: Add trackable
+    private func performAddTrackableEvent(_ event: Event.AddTrackableEvent) {
+        performAddOrTrack(event.trackable, resultHandler: event.resultHandler) {
```

Comment:
As above - we need to add `[weak self] in` if we call `self` inside the closure.

Reply:
As above - a `?` is missing

Reply:
Oops feb724012d3234dceea5ebbb18e3d84fd00ba23b